src/crawlee/static_content_crawler/_static_content_parser.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Generic
+
+from crawlee.static_content_crawler._static_crawling_context import TParseResult
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from crawlee.http_clients import HttpResponse
+
+
+@dataclass(frozen=True)
+class BlockedInfo:
+    """Information about whether the crawling is blocked. If reason is empty, then it means it is not blocked."""
+
+    reason: str
+
+    def __bool__(self) -> bool:
+        """No reason means no blocking."""
+        return bool(self.reason)
+
+
+class StaticContentParser(Generic[TParseResult], ABC):
+    """Parser used for parsing http response and inspecting parsed result to find links or detect blocking."""
+
+    @abstractmethod
+    async def parse(self, http_response: HttpResponse) -> TParseResult:
+        """Parse http response."""
+        ...
+
+    @abstractmethod
+    def is_blocked(self, parsed_content: TParseResult) -> BlockedInfo:
+        """Detect if blocked and return BlockedInfo with additional information."""
+        ...
```

Comment:
I think we should add a `match_selector` method to this base class and give `is_blocked` a default implementation that uses this new method. It will save us ~10 LoC :grinning: